pkg/game/elements/position.go

```diff
@@ -0,0 +1,65 @@
+package elements
+
+import (
+	"fmt"
+
+	sideMod "github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/side"
+)
+
+type Position struct {
+	// int8 would be fine for base game (72 tiles) but let's be a bit more generous
+	x int16
+	y int16
+}
+
+func NewPosition(x int16, y int16) Position {
+	return Position{x, y}
+}
+
+func (pos Position) X() int16 {
+	return pos.x
+}
+
+func (pos Position) Y() int16 {
+	return pos.y
+}
+
+func (pos Position) Add(other Position) Position {
+	return NewPosition(pos.x+other.x, pos.y+other.y)
+}
+
+/*
+Returns relative position directed by the side.
+Caution! It is supposed to be used with side directing only one cardinal direction (or two edges connected by corner)!
+Otherwise it will return undesired value!
```

Comment:
We might also want to handle the case when the function would return undesired value.

Reply:
TBH, I'm not a fan of this idea. It adds a lot of complexity to a very simple function.
Maybe that's just a premature optimisation thought, but I have a feeling iterating through all primary sides to check if more than one is set everytime we call this function would make the engine run slower.

Reply:
Well, I'm not a fan of methods that require more lines of documentation designated for how this function can fail than how it's actually used. This method incorrectly handles invalid cases and minimal solution would be to just test for strict equallity (i.e. side == sidemod.Top) so that the default case would be actually useful where one could panic or return identity. Test cases on how invalid cases are handled would also be welcomed.

Reply:
Well, I'm not a fan of methods that require more lines of documentation designated for how this function can fail than how it's actually used. This method incorrectly handles invalid cases and minimal solution would be to just test for strict equallity (i.e. side == sidemod.Top) so that the default case would be actually useful where one could panic or return identity. Test cases on how invalid cases are handled would also be welcomed.

Reply:
Added panic on incorrect input, and updated the documentation comment to better reflect what the function actually does